build/manager/Dockerfile-debug

```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.17 as builder
+FROM docker.io/golang:1.17 as builder
```

Comment:
Why is this needed?

Reply:
I get this prompt, which is undesirable, on Linux when I use `podman` instead of `docker`:

```sh
$ make docker-build-agent             
podman build --no-cache -t kusk-gateway-agent:dev --build-arg GOPROXY= -f ./build/agent/Dockerfile .
WARN[0000] missing "TELEMETRY_TOKEN" build argument. Try adding "--build-arg TELEMETRY_TOKEN=<VALUE>" to the command line 
[1/2] STEP 1/10: FROM golang:1.17 AS builder
? Please select an image: 
  ▸ registry.fedoraproject.org/golang:1.17
    registry.access.redhat.com/golang:1.17
    docker.io/library/golang:1.17
    quay.io/golang:1.17
```

As you can see, in the above, it finds the image in multiple registries which leads to a prompt.

In general, I think it's better to fully qualify the image name/path.

Reply:
Ah, i didn't know about this, given Ive never used Podman